src/libmdb/backend.c

```diff
@@ -155,6 +155,21 @@ enum {
 
 static void mdb_drop_backend(gpointer key, gpointer value, gpointer data);
 
+
+gchar *passthrough_unchanged(gchar *s) {
+    return s;
+}
+
+gchar *to_lower_case(gchar *s) {
+    for (gchar *p = s; *p; p++) *p = tolower(*p);
+    return s;
+}
+
+gchar *to_upper_case(gchar *s) {
+    for (gchar *p = s; *p; p++) *p = toupper(*p);
+    return s;
+}
```

Comment:
The case transformation functions will need to be UTF-8 compatible.
Looking at the g_utf8_strup set of functions as suggested by @evanmiller

Reply:
Done!